.github/workflows/build-and-push.yml

```diff
@@ -12,39 +12,68 @@ on:
       - '.tekton/**'
     tags:
       - v*
+  pull_request:
 
 jobs:
-  # Ensure that tests pass before publishing a new image.
-  build:
+  test:
     runs-on: ubuntu-latest
     env:
-      IMAGE_NAME: kserve/modelmesh-controller
       CI: true
     steps:
       - uses: actions/checkout@v2
       - name: Build develop image
         run: make build.develop
       - name: Run lint
         run: ./scripts/develop.sh make fmt
-      - name: Run unit tests
+      - name: Run unit test
         run: ./scripts/develop.sh make test
-      - name: Build runtime image
-        run: make build
-      - name: Log in to Docker Hub
-        run: docker login -u ${{ secrets.DOCKER_USER }} -p ${{ secrets.DOCKER_ACCESS_TOKEN }}
-      - name: Push to Docker Hub
+  build:
+    needs: test
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
```

Comment:
build not push on PRs https://github.com/kserve/rest-proxy/pull/22
```suggestion
```

Reply:
@ckadner